workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncher.cs

```diff
@@ -300,238 +315,89 @@ private async Task<bool> TriggerUploadAssemblyAsync()
-                            OutputRedirectBehaviour.RedirectStdErr)
-                        .InDirectory(DotNetWorkingDirectory)
-                        .RunAsync();
+                        simPlayerDeploymentName,
+                        Path.Combine(ProjectRootPath, simPlayerLaunchJson)
+                    });
+                }
 
-                    if (processResult.ExitCode == 0)
+                return DeploymentTasks.TriggerLaunchDeploymentAsync(arguments, () =>
+                {
+                    Application.OpenURL(string.Format(ConsoleURLFormat, projectName, deploymentName));
+                    if (simPlayerDeploymentEnabled)
                     {
-                        Application.OpenURL(string.Format(ConsoleURLFormat, projectName, deploymentName));
-                        if (simPlayerDeploymentEnabled)
-                        {
-                            Application.OpenURL(string.Format(ConsoleURLFormat, projectName, simPlayerDeploymentName));
-                        }
+                        Application.OpenURL(string.Format(ConsoleURLFormat, projectName, simPlayerDeploymentName));
                     }
-
-                    return processResult.ExitCode != 0;
-                }
-                finally
-                {
-                    EditorApplication.UnlockReloadAssemblies();
-                }
+                });
             }
 
-            private async Task<bool> TriggerStopDeploymentAsync(DeploymentInfo deployment)
+            private DeploymentTasks.WrappedTask<bool> TriggerStopDeploymentAsync(DeploymentInfo deployment)
             {
-                if (!Tools.Common.CheckDependencies())
-                {
-                    return false;
-                }
-
                 var arguments = new List<string>
                 {
                     "stop",
                     projectName,
                     deployment.Id
                 };
-                var processResult = await RunDeploymentLauncherHelperAsync(arguments, true);
-                if (processResult.ExitCode != 0)
-                {
-                    if (processResult.Stdout.Count > 0 && processResult.Stdout[0] == "<error:unknown-deployment>")
-                    {
-                        Debug.LogError(
-                            $"Unable to stop deployment {deployment.Name}. Has the deployment been stopped already?");
-                    }
-
-                    return false;
-                }
 
-                Debug.Log($"Deployment {deployment.Name} stopped.");
-                return true;
+                return DeploymentTasks.TriggerStopDeploymentAsync(arguments,
+                    () => { Debug.Log($"Deployment {deployment.Name} stopped."); },
+                    processResult =>
+                    {
+                        if (processResult.Stdout.Count > 0 && processResult.Stdout[0] == "<error:unknown-deployment>")
```

Comment:
let's not use magic strings for checking for errors. likewise for the other error string that you are checking for

Reply:
I didn't write this code - we can refactor this as a bigger pass over the deployment launcher. The UI and business logic in Unity needs to be separated and the underlying .NET program needs to be refactored as well